### PR TITLE
resolve #915: NextCloud guide: add more step info and flesh out descriptions

### LINKED ISSUE #915: [nextcloud] add more info to the described steps
hi there,

just having gone through the nice Nextcloud lab, I took the liberty to add more info to the described steps.

### PR DESCRIPTION
**Pull request by fl**
hi there, here's a PR for #915 , contributing to the NextCloud guide a bit

### CODE REVIEW

**Review by EV21 — COMMENTED** [review]
Very nice additions :bouquet:
I left some review comments for you.

**Comment by fl** [comment]
thanks @EV21 - fixed the typos as you suggested, marking them as resolved. Bonus :-) found one additional instance of referring to <version> not latest with a Nextcloud link

File: source/guide_nextcloud.rst [hunk 1]
```diff
@@ -82,7 +88,7 @@ To do that, create ``~/etc/php.d/opcache.ini`` with the content:
 PHP Memory
 ^^^^^^^^^^
 
-In order to increase the memory limit of php to the recommended minimum value of 512 MB, create ``~/etc/php.d/memory_limit.ini`` with the following content:
+NextCloud `recommends 512 MB <https://docs.nextcloud.com/server/20/admin_manual/installation/system_requirements.html>`_ of memory for an installation. To set the PHP memory limit to 512 MB, create the file ``~/etc/php.d/memory_limit.ini`` with the following content:
```
[the remark this answers is not in the document]
Comment (fl) [reply source/guide_nextcloud.rst]: done

File: source/guide_nextcloud.rst [hunk 2]
```diff
@@ -91,16 +97,13 @@ In order to increase the memory limit of php to the recommended minimum value of
 Output Buffering
 ^^^^^^^^^^^^^^^^
 
-Disable output buffering, create ``~/etc/php.d/output_buffering.ini`` with the following content:
+To Disable output buffering, create the file ``~/etc/php.d/output_buffering.ini`` with the following content:
```
[the remark this answers is not in the document]
Comment (fl) [reply source/guide_nextcloud.rst]: done

File: source/guide_nextcloud.rst [hunk 3]
```diff
@@ -126,29 +128,44 @@ Downloading
 
 Setup
 -----
-.. warning:: We strongly recommend you to use the MySQL backend for nextcloud. Do not use the SQLite backend for production. It gives you a better performance and reduces disk load on the host you share.
+.. warning:: We strongly recommend to use the MySQL backend for NextCloud, consistent with the NextCloud recommendation: `Using MariaDB/MySQL instead of SQLite <https://docs.nextcloud.com/server/20/admin_manual/installation/server_tuning.html#using-mariadb-mysql-instead-of-sqlite>`_. This yields better performance and reduces disk load on the host you share. Do not use the SQLite backend for production.
```
[the remark this answers is not in the document]
Comment (fl) [reply source/guide_nextcloud.rst]: done

File: source/guide_nextcloud.rst [hunk 4]
```diff
@@ -126,29 +128,44 @@ Downloading
...
 
-You need to change the following tree highlighted parameters to prepare the install command below:
+Create the database
+^^^^^^^^^^^^^^^^^
```
[the remark this answers is not in the document]
Comment (fl) [reply source/guide_nextcloud.rst]: done

File: source/guide_nextcloud.rst [hunk 5]
```diff
@@ -110,10 +113,9 @@ After that you need to restart PHP configuration to load the changes:
 
 Downloading
 -----------
+.. note:: This Lab uses a well-known link to the latest NextCloud package. You can find more versions and formats on the NextCloud `download page <https://nextcloud.com/install/#instructions-server>`_.
```
[the remark this answers is not in the document]
Comment (fl) [reply source/guide_nextcloud.rst]: done

File: source/guide_nextcloud.rst [hunk 6]
```diff
@@ -126,29 +128,44 @@ Downloading
...
-.. warning:: We strongly recommend you to use the MySQL backend for nextcloud. Do not use the SQLite backend for production. It gives you a better performance and reduces disk load on the host you share.
+.. warning:: We strongly recommend to use the MySQL backend for NextCloud, consistent with the NextCloud recommendation: `Using MariaDB/MySQL instead of SQLite <https://docs.nextcloud.com/server/20/admin_manual/installation/server_tuning.html#using-mariadb-mysql-instead-of-sqlite>`_. This yields better performance and reduces disk load on the host you share. Do not use the SQLite backend for production.
 
-You need to change the following tree highlighted parameters to prepare the install command below:
+Create the database
+^^^^^^^^^^^^^^^^^
 
- * Administrator username and password: Insert the credentials you want to use for the admin user
- * Get your :manual_anchor:`MySQL Password <database-mysql.html#login-credentials>`
+First, create a MySQL database to hold your NextCloud installation
```
[the remark this answers is not in the document]
Comment (fl) [reply source/guide_nextcloud.rst]: done

File: source/guide_nextcloud.rst [hunk 7]
```diff
@@ -126,29 +128,44 @@ Downloading
...
-  [isabell@stardust html]$ php occ maintenance:install --admin-user "${ADMIN_USER}" --admin-pass "${ADMIN_PASS}" --database 'mysql' --database-name "${USER}_nextcloud"  --database-user "${USER}" --database-pass "${MYSQL_PASSWORD}" --data-dir "${HOME}/nextcloud_data"
+  [isabell@stardust html]$ php occ maintenance:install --admin-user "${NEXTCLOUD_ADMIN_USER}" --admin-pass "${NEXTCLOUD_ADMIN_PASS}" --database 'mysql' --database-name "${USER}_nextcloud"  --database-user "${USER}" --database-pass "${MYSQL_PASSWORD}" --data-dir "${HOME}/nextcloud_data"
   Nextcloud was successfully installed
   [isabell@stardust html]$
 
-You should now set your :manual:`domain <web-domains>`.
+Set the "truste" domain
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Set the :manual:`domain name <web-domains>` that your NextCloud will be accessed with as "trusted":
```
[the remark this answers is not in the document]
Comment (fl) [reply source/guide_nextcloud.rst]: done

File: source/guide_nextcloud.rst [hunk 8]
```diff
@@ -126,29 +128,44 @@ Downloading
...
   --------------
+  [isabell@stardust html]$
+
+Run the installation script
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now, execute the NextCloud maintenance PHP script ``occ`` with the parameters shown below. Set the three highlighted parameters to your account-specific values before executing the command:
+
+* ``NEXTCLOUD_ADMIN_USER`` the name of the NextCloud admin user you have to use when logging on for the first time on your new NextCloud Instance
+* ``NEXTCLOUD_ADMIN_PASS`` the password for the NextCloud admin user
```
[the remark this answers is not in the document]
Comment (fl) [reply source/guide_nextcloud.rst]: done

File: source/guide_nextcloud.rst [hunk 9]
```diff
@@ -301,6 +322,7 @@ If the web installation fails, install the app manually in your shell:
 
   [isabell@stardust html]$ cd apps
   [isabell@stardust apps]$ curl -L https://github.com/nextcloud/spreed/releases/download/v8.0.7/spreed-8.0.7.tar.gz | tar -xvzf -
+  [isabell@stardust html]$
```
[the remark this answers is not in the document]
Comment (fl) [reply source/guide_nextcloud.rst]: done

**Comment by fl** [comment]
@EV21 - just so you know, I'm adding some Nextcloud references, going section by section https://github.com/fl/lab/tree/nc-guide-part2 but I'm not done yet.

**Comment by fl** [comment]
anything missing?

**Comment by EV21** [comment]
With your (deleted) comment, it looked like you wanted to add your other commit to this pull request. https://github.com/fl/lab/commit/c07c5c282ed43d0821318e8acb6de8e77989954d
So you want to do this in two parts and this PR is done?

**Comment by luto** [comment]
Thank you! :tada: and thanks for the review @EV21!

**Pull request merged**